### PR TITLE
Issue 228 implement hash code method tests config

### LINKED ISSUE #228: Implement hashCode() method tests for config classes

### PR DESCRIPTION
**Pull request by f-porter**
Caution: Please merge #226 first as this PR depends on it!
Closes #228.

### CODE REVIEW

**Comment by thechauffeur** [comment]
Please rebase this PR @f-porter because commits which are already merged are in here.

**Comment by thechauffeur** [comment]
Do not review this PR prior to the rebase.

**Comment by f-porter** [comment]
I rebased it.

File: src/test/java/de/qabel/core/ExtendedHashCodeMethodTester.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
+package de.qabel.core;
```
Comment (aBothe) [inline src/test/java/de/qabel/core/ExtendedHashCodeMethodTester.java]: Imho, all these helper classes (including the factories that are dedicated for testing purposes only) should be in some of the test package/namespaces - if there are any.

**Comment by aBothe** [comment]
Generally looks fine to me. Unforeseen semantic problems may be fixed later on anyway :P

File: src/test/java/de/qabel/core/ExtendedHashCodeMethodTester.java (hunk 1 above)
Comment (f-porter) [inline src/test/java/de/qabel/core/ExtendedHashCodeMethodTester.java]: Actually there is no test namespace. The tests for e.g. the config package are in the same package as the config (obviously for easier access).
I chose this package for the helper class as there will be other additional subpackages where this class could be needed.

**Comment by k-c13** [comment]
Looks good

**Comment by jan-schreib** [comment]
As soon as travis passes, I give this pullreq a full ack.

**Comment by thechauffeur** [comment]
@f-porter: @jan-schreib and I reviewed this and would ACK this PR but after checking locally (which didn't work) we re-triggered travis and two (new) tests fail:
```
de.qabel.core.config.ConfigHashCodeTest > storageVolumeHashCodeTest FAILED
    org.meanbean.factories.NoSuchFactoryException at ConfigHashCodeTest.java:60
        Caused by: org.meanbean.factories.ObjectCreationException at ConfigHashCodeTest.java:60
            Caused by: org.meanbean.factories.NoSuchFactoryException at ConfigHashCodeTest.java:60
                Caused by: org.meanbean.factories.ObjectCreationException at ConfigHashCodeTest.java:60
                    Caused by: java.lang.NoSuchMethodException at ConfigHashCodeTest.java:60

de.qabel.core.config.ConfigHashCodeTest > localSettingsHashCodeTest FAILED
    java.lang.AssertionError at ConfigHashCodeTest.java:102
```

**Comment by f-porter** [comment]
I merged master into this branch and fixed the tests. Should be fine now.

**Comment by jasinai** [comment]
Looks good: :+1:

**Comment by thechauffeur** [comment]
I ACK the last two commits.

**Comment by jan-schreib** [comment]
I looked at the merge commit with `git show` everything is fine. :+1:

**Pull request merged**